account_invoice_validation/models/account_move.py

```diff
@@ -0,0 +1,450 @@
+    )
+
+    note = fields.Char(
+        help="Reason of the last modification of the state",
+    )
+
+    can_edit_validation_user = fields.Boolean(
+        help="Technical field: true if user can edit approver user",
+        compute="_compute_can_edit_validation_user",
+    )
+
+    approved_once = fields.Boolean(
+        help="Technical field: when using 2 approbations, set to true after first approbation",
+        compute="_compute_validation_user_id",
+        store=True,
+        copy=False,
+    )
+
+    @api.onchange("partner_id")
+    def _onchange_validation_user_id(self):
+        use_invoice_first_approval = (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("account_invoice_validation.use_invoice_first_approval")
+        )
+        if use_invoice_first_approval:
+            for rec in self:
+                val_user_id = rec.partner_id.validation_user_id
+                rec.validation_user_id = (
+                    val_user_id or rec.company_id.validation_user_id
+                )
+
+    @api.depends("move_type", "partner_id")
+    def _compute_validation_user_id(self):
+        use_invoice_first_approval = (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("account_invoice_validation.use_invoice_first_approval")
+        )
+
+        for rec in self:
+            if rec.move_type in rec.get_concerned_types():
+                val_user_id = self.env["res.partner"]
+                approved_once = False
+                # try to take approver on partner
+                if use_invoice_first_approval:
+                    val_user_id = rec.partner_id.validation_user_id
+
+                # no approver set
+                # only one approbation is needed
+                # 2 cases:
+                #  - validation user is not set
+                #  - parameter not activated
+                if not val_user_id:
+                    approved_once = True
+
+                val_user_id = val_user_id or rec.company_id.validation_user_id
+
+                rec.write(
+                    {
+                        "validation_user_id": val_user_id,
+                        "approved_once": approved_once,
+                    }
+                )
+
+    @api.depends_context("uid")
+    def _compute_can_edit_validation_user(self) -> None:
+        self.can_edit_validation_user = self.env.user.has_group(
+            "account_invoice_validation.group_account_invoice_validation_assign"
+        )
+
+    @api.depends("validation_user_id")
+    def _compute_validation_state(self) -> None:
+        self.filtered(
+            lambda rec: rec.validation_user_id and not rec.validation_state
+        ).update({"validation_state": "wait_approval"})
+
+    @api.depends("validation_user_id")
+    def _compute_date_assignation(self) -> None:
+        self.filtered(lambda rec: rec.validation_user_id).update(
+            {"date_assignation": fields.Date.today()}
```

Comment:
You don't set a value for records without `validation_user_id`. That's not correct and could raise an exception in some case.

That's the same for `_compute_validation_state(...)` but in this field you set a `default=""` so it's correct.